python/ray/_private/runtime_env/context.py

```diff
@@ -67,23 +70,15 @@ def exec_worker(self, passthrough_args: List[str], language: Language):
             class_path_args = ["-cp", ray_jars + ":" + str(":".join(local_java_jars))]
             passthrough_args = class_path_args + passthrough_args
         elif language == Language.JULIA:
-            executable = "julia"
-            args = [
-                "-e", "'using Ray; start_worker()'",
-                "--"
-            ]
-            # TODO(omus): required to avoid escaping the Julia code. Ideally
-            # this information would be passed in via the serialized runtime
-            # context.
-            executable = " ".join([executable] + args)
+            command = self.julia_command or ["julia", "-e", "using Ray; start_worker()"]
+            command += ["--"]
         elif sys.platform == "win32":
-            executable = ""
+            command = [""]
         else:
-            executable = "exec "
+            command = ["exec"]
 
-        passthrough_args = [s.replace(" ", r"\ ") for s in passthrough_args]
-        exec_command = " ".join([f"{executable}"] + passthrough_args)
-        command_str = " ".join(self.command_prefix + [exec_command])
+        command = self.command_prefix + command + passthrough_args
+        command_str = shlex.join(command)
```

Comment:
nice, had to look this up but it does seem to be what we want here (and a nice improvement to upstream!)